### DISCUSSION TITLE
Question about parameter x in learn_one of DBSTREAM

### DISCUSSION THREAD

**Discussion by sapfelaft** [online-ml/river#921] [Q&A]
Hey,
I wonder what the expected input for learn_one of DBSTREAM is? Should it be a precomputed distance matrix (like cosine_distances)? Or should I fit "raw" vectors on DBSTREAM?

When I fit tfidf vectors on DBSTREAM, the result is imprecise (or to put it more precisely: absolutely wrong). But if I fit a precomputed cosine distance matrix, the result is quite good, even without changing default parameters. 

I have taken a closer look at the code, and what I found is that the minkowski/euclidean distance is used for distance calculation. In my opinion the huge difference in results of raw tfidf/cosine distances makes no sense when euclidean distances are calculated anyway (what is the euclidean distance of cosine distances?).

Can you provide some information? Perhaps I am on the wrong track.

**Comment by MaxHalford** [comment]
Hey @hoanganhngo610, could you answer this one if you have some time?
